### PR TITLE
mw.config returns type any

### PR DESCRIPTION
**Pull request by jdlrobson**

### CODE REVIEW

**Review by DannyS712 — CHANGES_REQUESTED** [review]
(github won't let me leave an inline comment on the line since its not a part of the patch, so putting here) the same applies to mw.user.options because that also uses a `Map` object to store things, and that should also be updated to return any - or maybe instead of duplicating the documentation we should create a `Map` interface object that both of them use. -1

**Comment by jdlrobson** [comment]
> (github won't let me leave an inline comment on the line since its not a part of the patch, so putting here) the same applies to mw.user.options because that also uses a `Map` object to store things, and that should also be updated to return any - or maybe instead of duplicating the documentation we should create a `Map` interface object that both of them use. -1

Good idea. Done!

File: MediaWiki.d.ts [hunk 1]
```diff
@@ -69,6 +66,10 @@ interface MwExperiments {
 	getBucket( bucket: MwExperimentBucket, token: string ): string;
 }
 
+interface MwMap {
```
Comment (DannyS712) [inline MediaWiki.d.ts:69]: we can also add this as a direct part of the MediaWiki interface, available as `mw.Map` client side.

**Review by DannyS712 — COMMENTED** [review]
Suggest changing commit message to be "Add and use mw.Map interface" or similar since this isn't just about mw.config anymore

File: MediaWiki.d.ts (hunk 1 above)
Comment (jdlrobson) [reply MediaWiki.d.ts:69]: Done.

**Comment by jdlrobson** [comment]
> Suggest changing commit message to be "Add and use mw.Map interface" or similar since this isn't just about mw.config anymore

Yeh I do that on merge in Githubs workflow.

**Pull request merged**